src/components/App.js

```diff
@@ -5,11 +5,16 @@ import MapContainer from './MapContainer'
 import Sidebar from './Sidebar'
 import DevTools from './DevTools'
 import store from '../store'
+import { initApp } from '../url-state'
 
 import 'semantic-ui-css/semantic.min.css'
 import './App.css'
 
 class App extends Component {
+  constructor (props) {
+    super(props)
+    initApp()
```

Comment:
let's actually put `initApp` outside of React entirely! You can call this in `src/index.js` before the `ReactDOM.render()`. State would be ready before the component even gets called.